Modules/Core/Common/test/itkSobelOperatorImageConvolutionTest.cxx

```diff
@@ -0,0 +1,83 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkConstNeighborhoodIterator.h"
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkImageRegionIterator.h"
+#include "itkNeighborhoodInnerProduct.h"
+#include "itkSobelOperator.h"
+#include "itkTestingMacros.h"
+
+
+int
+itkSobelOperatorImageConvolutionTest(int argc, char * argv[])
+{
+  if (argc != 4)
+  {
+    std::cerr << "Missing parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv) << " inputFileName direction outputFileName"
+              << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 2;
+
+  using PixelType = unsigned char;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  using SobelOperatorType = itk::SobelOperator<PixelType, Dimension>;
+
+  using NeighborhoodIteratorType = itk::ConstNeighborhoodIterator<ImageType>;
+  using IteratorType = itk::ImageRegionIterator<ImageType>;
+
+
+  const auto inputImage = itk::ReadImage<ImageType>(argv[1]);
+
+
+  SobelOperatorType sobelOperator;
+
+  auto direction = std::stoul(argv[2]);
+  sobelOperator.SetDirection(direction);
+
+  itk::Size<Dimension> radius;
+  radius.Fill(1);
+  sobelOperator.CreateToRadius(radius);
+
+  NeighborhoodIteratorType it(radius, inputImage, inputImage->GetRequestedRegion());
```

Comment:
Maybe change 'requested region' to 'largest possible region' here and below? This test randomly fails, e.g.
https://open.cdash.org/test/862523065
https://open.cdash.org/test/862546031
https://open.cdash.org/test/862426223
and many more.
Not tested and not 100% sure.
cc @dzenanz

Reply:
Thanks for reporting @issakomi :+1:.

Not sure if the region might be the cause. I realize that I used the same names for the output files of the two different tests `itkSobelOperatorImageFilterTest` and `itkSobelOperatorImageConvolutionTest`: maybe the issue is that tests clash with each other. Despite the reported failures only related to the horizontal detector. I'll propose a patch set to use different output names for each test and see if that solves it.

Reply:
PR #3746. Resolving.